guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,6 @@
+## Performance related
+
+* Avoid allocations in hot paths:
+* Avoid LINQ.
```

Comment:
I'd say avoid `System.Linq` but also let's limit this to hot paths as well, as we're good with linq usage in non-hot paths so far.